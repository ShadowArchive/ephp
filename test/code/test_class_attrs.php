<?php

class Person {
    private $name;
    private $surname;

    public function __construct($name, $surname) {
        $this->name = $name;
        $this->surname = $surname;
    }

    public function getName() { return $this->name; }
    public function getSurname() { return $this->surname; }

    public function __toString() {
        return "My name is {$this->surname}... {$this->name} {$this->surname}\n";
    }
}

$p = new Person("Manuel", "Rubio");
print $p;
print $p->name;
print $p->surname;
print "OK\n";
