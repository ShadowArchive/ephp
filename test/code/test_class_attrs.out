My name is Rubio... Manuel Rubio

Fatal error: Cannot access private property Person::$name in {{CWD}}/test/code/test_class_attrs.php on line 22
